apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestPlugLatencyActivity.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright 2019 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mobileer.oboetester;
+
+import android.annotation.TargetApi;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.media.AudioDeviceCallback;
+import android.media.AudioDeviceInfo;
+import android.media.AudioManager;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.mobileer.audio_device.AudioDeviceInfoConverter;
+
+import java.io.IOException;
+import java.util.HashMap;
+
+/**
+ * Tests the latency of plugging in or unplugging an audio device.
+ */
+public class TestPlugLatencyActivity extends TestAudioActivity {
+
+    public static final int POLL_DURATION_MILLIS = 1;
+
+    private TextView     mInstructionsTextView;
+    private TextView     mPlugTextView;
+    private TextView     mAutoTextView;
+    MyAudioDeviceCallback mDeviceCallback = new MyAudioDeviceCallback();
+    private AudioManager mAudioManager;
+
+    private volatile int mPlugCount = 0;
+
+    class MyAudioDeviceCallback extends AudioDeviceCallback {
+        private HashMap<Integer, AudioDeviceInfo> mDevices
+                = new HashMap<Integer, AudioDeviceInfo>();
+
+        @Override
+        public void onAudioDevicesAdded(AudioDeviceInfo[] addedDevices) {
```

Comment:
Ignore the initial flurry of devices added, which includes speakers. Just log the add/remove messages.

Reply:
Thanks!